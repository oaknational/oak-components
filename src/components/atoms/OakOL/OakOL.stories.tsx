import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakOL, OakOLProps } from "./OakOL";

import { OakLI } from "@/components/atoms/OakLI";
import { typographyArgTypes } from "@/storybook-helpers/typographyStyleHelpers";

/**
 *
 * OakOL is a styled ol tag.
 * Use the controls to view different font styles.
 *
 */

const meta: Meta<typeof OakOL> = {
  component: OakOL,
  tags: ["autodocs"],
  title: "components/atoms/OakOL",
  argTypes: {
    ...typographyArgTypes,
  },
  parameters: {
    controls: {
      include: [...Object.keys(typographyArgTypes)],
      sort: "none",
    },
  },
};
export default meta;

type Story = StoryObj<typeof OakOL>;

export const DefaultOakOL: Story = (args: Partial<OakOLProps>) => (
  <OakOL data-testId="OakOL-id" {...args}>
    <OakLI>Item 1</OakLI>
    <OakLI>Item 2</OakLI>
    <OakLI>Item 3</OakLI>
  </OakOL>
);

DefaultOakOL.args = { $font: "list-item-1" };
