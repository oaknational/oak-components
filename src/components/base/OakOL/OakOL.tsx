import styled from "styled-components";

import { ColorProps, MarginProps, colorStyle, marginStyle } from "@/styles";

export type OakOLProps = MarginProps & ColorProps;
/**
 * Styled `ol` (ordered list) component.
 *
 * ## Usage
 *
 * Use where we have an ordered list to ensure numbers are styled
 *
 * */
const OakOL = styled.ol<OakOLProps>`
  ${marginStyle}
  ${colorStyle}
  counter-reset: item;
  padding: 0;

  li {
    display: block;
    counter-increment: item;
    margin: 0;
    padding: 0 0 0 16px;
    text-indent: -16px;
    list-style-type: none;
    line-height: 32px;

    // Portable text generates linebreaks within list items

    br {
      content: "";
      display: block;
      margin-top: 8px;
    }
  }

  & li::before {
    padding-right: 4px;
    content: counter(item) ".";
  }
  a {
    color: ${(props) =>
      props.theme &&
      props.theme.uiColors &&
      props.theme.uiColors["text-link-active"]};
  }w
`;

export default OakOL;
