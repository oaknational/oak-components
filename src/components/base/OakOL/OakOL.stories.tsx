import React from "react";
import { Meta, StoryObj } from "@storybook/react";

import { OakOL, OakOLProps } from "../OakOL/OakOL";
import { OakLI } from "../OakLI/OakLI";

import { typographyArgTypes } from "@/storybook-helpers/typographyStyleHelpers";

/**
 *
 * OakOL is a styled ol tag.
 * Use the controls to view different font styles.
 *
 */

const meta: Meta<typeof OakOL> = {
  component: OakOL,
  tags: ["autodocs"],
  title: "components/base/OakOL",
  argTypes: {
    ...typographyArgTypes,
  },
  parameters: {
    controls: {
      include: [...Object.keys(typographyArgTypes)],
      sort: "none",
    },
  },
};
export default meta;

type Story = StoryObj<typeof OakOL>;

export const DefaultOakOL: Story = (args: Partial<OakOLProps>) => (
  <OakOL data-testId="OakOL-id" {...args}>
    <OakLI>Item 1</OakLI>
    <OakLI>Item 2</OakLI>
    <OakLI>Item 3</OakLI>
  </OakOL>
);

DefaultOakOL.args = { $font: "list-item-1" };
