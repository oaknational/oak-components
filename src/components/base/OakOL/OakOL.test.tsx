import React from "react";
import "@testing-library/jest-dom";
import { create } from "react-test-renderer";

import OakOL from "./OakOL";

import renderWithTheme from "@/test-helpers/renderWithTheme";

describe("Component OakOL", () => {
  it("renders", () => {
    const { getByTestId } = renderWithTheme(<OakOL data-testid="test" />);
    expect(getByTestId("test")).toBeInTheDocument();
  });
  it("matches snapshot", () => {
    const tree = create(<OakOL />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
