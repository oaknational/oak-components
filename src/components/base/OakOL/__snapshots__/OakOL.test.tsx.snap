// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component OakOL matches snapshot 1`] = `
<ol
  className="sc-aXZVg FWJxw"
/>
`;
